ehrapy/_utils_data.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from anndata import AnnData
+
+
+def are_dataset_equal(arr1: np.ndarray, arr2: np.ndarray) -> np.bool_:
```

Comment:
```suggestion
def are_ndarrays_equal(arr1: np.ndarray, arr2: np.ndarray) -> np.bool_:
```

This would much better reflect the current behavior. I wouldn't mind having this completely generalized though with all matrix types.